From ae64dcfa6b83d5ee9a4ff0c7ae44c0e848e4d57b Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:08:54 -0400
Subject: [usb] usb_wwan: report failed submissions as errors

Message-id: <1410887463-20674-463-git-send-email-dzickus@redhat.com>
Patchwork-id: 95643
O-Subject: [RHEL7 PATCH 462/591] USB: usb_wwan: report failed submissions as errors
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit 8bb7ec65d600fd513aa94b50078a6329df612daa
Author: Johan Hovold <jhovold@gmail.com>
Date:   Mon May 26 19:23:31 2014 +0200

    USB: usb_wwan: report failed submissions as errors

    Promote failed-submission messages in open() and write() to error log
    level.

    Signed-off-by: Johan Hovold <jhovold@gmail.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/serial/usb_wwan.c b/drivers/usb/serial/usb_wwan.c
index c951f75..bbcbdaa 100644
--- a/drivers/usb/serial/usb_wwan.c
+++ b/drivers/usb/serial/usb_wwan.c
@@ -231,9 +231,9 @@ int usb_wwan_write(struct tty_struct *tty, struct usb_serial_port *port,
    spin_unlock_irqrestore(&intfdata->susp_lock, flags);
    err = usb_submit_urb(this_urb, GFP_ATOMIC);
    if (err) {
-    dev_dbg(&port->dev,
-     "usb_submit_urb %p (write bulk) failed (%d)\n",
-     this_urb, err);
+    dev_err(&port->dev,
+     "%s: submit urb %d failed: %d\n",
+     __func__, i, err);
     clear_bit(i, &portdata->out_busy);
     spin_lock_irqsave(&intfdata->susp_lock, flags);
     intfdata->in_flight--;
@@ -376,7 +376,7 @@ int usb_wwan_open(struct tty_struct *tty, struct usb_serial_port *port)
  if (port->interrupt_in_urb) {
   err = usb_submit_urb(port->interrupt_in_urb, GFP_KERNEL);
   if (err) {
-   dev_dbg(&port->dev, "%s: submit int urb failed: %d\n",
+   dev_err(&port->dev, "%s: submit int urb failed: %d\n",
     __func__, err);
   }
  }
@@ -388,8 +388,9 @@ int usb_wwan_open(struct tty_struct *tty, struct usb_serial_port *port)
    continue;
   err = usb_submit_urb(urb, GFP_KERNEL);
   if (err) {
-   dev_dbg(&port->dev, "%s: submit urb %d failed (%d) %d\n",
-    __func__, i, err, urb->transfer_buffer_length);
+   dev_err(&port->dev,
+    "%s: submit read urb %d failed: %d\n",
+    __func__, i, err);
   }
  }
 
-- 
1.7.1